apps/backend/test/unit/reconciliation/pos-reconciliation.spec.ts

```diff
@@ -410,4 +412,83 @@ describe('PosReconciliationService', () => {
       ).toBe(true);
     });
   });
+  describe('heuristic round four', () => {
+    beforeAll(() => {
+      // create some data that will not match
+      const data = new MockData(PaymentMethodClassification.POS);
+      payments = data.paymentsMock as PaymentEntity[];
+      deposits = data.depositsMock as POSDepositEntity[];
+      /*eslint-disable */
+
+      const aggregatedPayments = service.aggregatePayments(
+        payments.filter((payment) =>
+          [MatchStatus.PENDING, MatchStatus.IN_PROGRESS].includes(
+            payment.status
+          )
+        )
+      );
+
+      const aggregatedDeposits = service.aggregateDeposits(
+        deposits
+      ) as unknown as POSDepositEntity[];
+      const matchedRoundFour = service.matchPosPaymentToPosDepositsRoundFour(
+        aggregatedPayments,
+        service.buildPosDepositsDictionary(aggregatedDeposits),
+        PosHeuristicRound.FOUR
+      );
+
+      roundFourPaymentMatches = matchedRoundFour.flatMap((itm) => itm.payments);
+      roundFourDepositMatches = matchedRoundFour.flatMap((itm) => itm.deposits);
+    });
+    it('should aggregate the deposits and build the deposit dictionary', () => {
+      const aggregatedDeposits = service.aggregateDeposits(
+        deposits
+      ) as unknown as POSDepositEntity[];
+
+      const depositsDictionary =
+        service.buildPosDepositsDictionary(aggregatedDeposits);
+      console.log(depositsDictionary);
+
+      const dictKeys = Object.keys(depositsDictionary);
+
+      expect(dictKeys).toEqual(
+        expect.arrayContaining(
+          aggregatedDeposits.map((itm) => itm.transaction_amt.toString())
+        )
+      );
+    });
+
+    it('should match based on date, aggregated amount, method ', () => {
+      console.log(service.aggregatePayments(roundFourPaymentMatches));
+      console.log(service.aggregateDeposits(roundFourDepositMatches));
+      const reaggregatedPayemnts = service.aggregatePayments(
+        roundFourPaymentMatches
+      );
+      const reaggregatedDeposits = service.aggregateDeposits(
+        roundFourDepositMatches
+      );
+      expect(
+        reaggregatedPayemnts
+          .map(
+            (payment) =>
```

Comment:
Might suggest being explicit and naming these inner variables `aggregatedPayment` and `aggregatedDeposit`. We want to be clear in avoiding confusion between payments and aggregated payments